### PR TITLE
Add e2e tests for Mixer Dashboard

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR follows-on the PR that setup e2e testing for the Istio grafana dashboard. It
establishes a set of baseline tests that verify that some data is available for each
of the queries in the Mixer dashboard.

This work is part of the effort to move the Mixer dashboard status from dev into alpha (and
eventually beta and stable) status.

As part of this effort, the existing Mixer dashboard components that referenced the old runtime
metrics were updated to match the new runtime2 metrics, and the barely supported rows feature
of grafana were removed where applicable.

mixerdashboard

Fixes: #2163

### CODE REVIEW

**Comment by douglas-reid** [comment]
Reviewers: please note the CircleCI build named e2e_dashboard (and the green status).

**Comment by jeffmendoza** [comment]
Please run `/addons/grafana/fix_datasources.sh`. See https://github.com/istio/istio/blob/master/addons/grafana/README.md#implementation-details for details.

The Grafana folks said they are working on something (https://community.grafana.com/t/what-is-the-correct-way-to-save-dashboard-json-for-use-in-provisioning/5254). I haven't checked the new 5.0 (non-beta) to see if this is improved. I'll do that.

**Comment by douglas-reid** [comment]
@jeffmendoza Script run.

File: tests/e2e/tests/dashboard/dashboard_test.go [hunk 1]
```diff
@@ -71,53 +74,53 @@ func TestMain(m *testing.M) {
 	os.Exit(tc.RunTest(m))
 }
 
-func TestIstioDashboard(t *testing.T) {
-	promAPI, err := promAPI()
-	if err != nil {
-		t.Fatalf("Could not build prometheus API client: %v", err)
-	}
+func TestDashboards(t *testing.T) {
+	// dashboards := map[string]string{"Istio": istioDashboard, "Mixer": mixerDashboard}
```
Comment (mandarjog) [inline tests/e2e/tests/dashboard/dashboard_test.go]: super nit: This is redundant.

File: tests/e2e/tests/dashboard/dashboard_test.go [hunk 2]
```diff
@@ -137,7 +140,7 @@ func sendTrafficToCluster() (*fhttp.HTTPRunnerResults, error) {
 	return fhttp.RunHTTPTest(&opts)
 }
 
-func sendTCPTrafficToCluster(t *testing.T) error {
+func sendTCPTrafficToCluster() error {
```
Comment (mandarjog) [inline tests/e2e/tests/dashboard/dashboard_test.go]: nit: `generateTCPInCluster()`

File: tests/e2e/tests/dashboard/dashboard_test.go [hunk 3]
```diff
@@ -307,5 +362,5 @@ func getPodList(namespace string, selector string) ([]string, error) {
 }
 
 func allowPrometheusSync() {
-	time.Sleep(30 * time.Second)
+	time.Sleep(1 * time.Minute)
```
Comment (mandarjog) [inline tests/e2e/tests/dashboard/dashboard_test.go]: nit: in the interest of sleeping as little as possible, can we have a polling loop here instead with a much smaller sleep?

File: tests/e2e/tests/dashboard/dashboard_test.go (hunk 1 above)
Comment (douglas-reid) [reply tests/e2e/tests/dashboard/dashboard_test.go]: Removed comment.

File: tests/e2e/tests/dashboard/dashboard_test.go (hunk 2 above)
Comment (douglas-reid) [reply tests/e2e/tests/dashboard/dashboard_test.go]: done.

File: tests/e2e/tests/dashboard/dashboard_test.go (hunk 3 above)
Comment (douglas-reid) [reply tests/e2e/tests/dashboard/dashboard_test.go]: Possibly. The issue is trying to figure out when prometheus successfully scrapes the cAdvisor data for the Mixer container -- the existence of that metric is what is being checked for in the dashboard test itself...  Let's punt on this for the moment.

**Comment by douglas-reid** [comment]
PTAL. Comments addressed.

**Pull request merged**